glib-build-tools/src/lib.rs

```diff
@@ -57,3 +62,73 @@ pub fn compile_resources<P: AsRef<Path>>(source_dirs: &[P], gresource: &str, tar
         println!("cargo:rerun-if-changed={dep}");
     }
 }
+
+// rustdoc-stripper-ignore-next
+/// Call to run `glib-compile-schemas` to generate compiled gschemas from `.gschema.xml` schemas
+/// files from specified directories and store `gschemas.compiled` into `glib-2.0` schema cache
+/// directory.
+///
+/// If `target_dir` is `None`, the default schema cache directory is used:
+/// - Unix: `$HOME/.local/share/glib-2.0/schemas/`
+/// - Windows: `C:/ProgramData/glib-2.0/schemas/`
+///
+/// ```no_run
+/// glib_build_tools::compile_schemas(
+///     &["schemas"],
+///     None
+/// );
+/// ```
+pub fn compile_schemas(schemas_dir: &[&str], target_dir: Option<&str>) {
+    let target_dir = match target_dir {
+        Some(base) => PathBuf::from(base),
+        None => {
+            let prefix = if cfg!(windows) {
+                PathBuf::from("C:/ProgramData")
+            } else {
+                glib::user_data_dir()
+            };
+
+            Path::new(&prefix).join("glib-2.0").join("schemas")
+        }
+    };
+
+    // Ensure target_dir exists
+    std::fs::create_dir_all(&target_dir).expect("Failed to create target directory");
```

Comment:
Installing things as part of `build.rs` is an anti-pattern and unexpected. You should not place (or modify) anything outside of `OUT_DIR`, see https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts